src/main/java/com/github/dariobalinzo/filter/WhitelistFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.github.dariobalinzo.filter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+public class WhitelistFilter implements Filter {
+
+    List<String> allowedValues;
+
+    public WhitelistFilter(List<String> allowedValues) {
+        this.allowedValues = allowedValues;
+    }
+
+    public List<String> getAllowedValues() {
+        return allowedValues;
+    }
+
+    public void setAllowedValues(List<String> allowedValues) {
+        this.allowedValues = allowedValues;
+    }
+
+    @Override
+    public Map<String, Object> filter(Map<String, Object> document) {
+        return document.entrySet().stream()
+                .filter(key -> allowedValues.contains(key.getKey()))
```

Comment:
to seep up `allowedValues.contains(key.getKey())` you can also use a Set instead of List